api/v1alpha08/sonataflow_types.go

```diff
@@ -24,6 +24,535 @@ import (
+	// +listType=map
+	// +listMapKey=name
+	// +featureGate=DynamicResourceAllocation
+	// +optional
+	ResourceClaims []corev1.PodResourceClaim `json:"resourceClaims,omitempty" patchStrategy:"merge,retainKeys" patchMergeKey:"name" protobuf:"bytes,39,rep,name=resourceClaims"`
+}
+
+func (f *FlowPodSpec) ToPodSpec() corev1.PodSpec {
+	return corev1.PodSpec{
+		Volumes:                       f.Volumes,
+		InitContainers:                f.InitContainers,
+		Containers:                    f.Containers,
+		RestartPolicy:                 f.RestartPolicy,
+		TerminationGracePeriodSeconds: f.TerminationGracePeriodSeconds,
+		ActiveDeadlineSeconds:         f.ActiveDeadlineSeconds,
+		DNSPolicy:                     f.DNSPolicy,
+		NodeSelector:                  f.NodeSelector,
+		ServiceAccountName:            f.ServiceAccountName,
+		AutomountServiceAccountToken:  f.AutomountServiceAccountToken,
+		NodeName:                      f.NodeName,
+		HostNetwork:                   f.HostNetwork,
+		HostPID:                       f.HostPID,
+		HostIPC:                       f.HostIPC,
+		ShareProcessNamespace:         f.ShareProcessNamespace,
+		SecurityContext:               f.SecurityContext,
+		ImagePullSecrets:              f.ImagePullSecrets,
+		Hostname:                      f.Hostname,
+		Subdomain:                     f.Subdomain,
+		Affinity:                      f.Affinity,
+		SchedulerName:                 f.SchedulerName,
+		Tolerations:                   f.Tolerations,
+		HostAliases:                   f.HostAliases,
+		PriorityClassName:             f.PriorityClassName,
+		Priority:                      f.Priority,
+		DNSConfig:                     f.DNSConfig,
+		ReadinessGates:                f.ReadinessGates,
+		RuntimeClassName:              f.RuntimeClassName,
+		EnableServiceLinks:            f.EnableServiceLinks,
+		PreemptionPolicy:              f.PreemptionPolicy,
+		Overhead:                      f.Overhead,
+		TopologySpreadConstraints:     f.TopologySpreadConstraints,
+		SetHostnameAsFQDN:             f.SetHostnameAsFQDN,
+		OS:                            f.OS,
+		HostUsers:                     f.HostUsers,
+		SchedulingGates:               f.SchedulingGates,
+		ResourceClaims:                f.ResourceClaims,
+	}
+}
+
+// FlowPodSpecTemplate describes the desired custom Kubernetes PodSpecTemplate definition for the deployed flow.
+//
+// The FlowContainer describes the container where the actual flow is running. It will override any default definitions.
+// For example, to override the image one can use `.spec.podSpecTemplate.flowContainer.image = my/image:tag`.
+type FlowPodSpecTemplate struct {
+	// FlowContainer is the Kubernetes container where the workflow application should run.
+	// One can change this attribute in order to override the defaults provided by the operator.
+	// +optional
+	FlowContainer   FlowContainer `json:"flowContainer,omitempty"`
+	PodSpecTemplate FlowPodSpec   `json:",omitempty"`
```

Comment:
Following analogy with the Kubernetes API -> PodTemplateSpec type, I think this filed "PodSpecTemplate" can be named just "PodSpec"

Reply:
Silly question here to double check if my understanding is correct, so FlowContainer FlowContainer can be eventually fine tuned by the users at the time they define their deployment. And PodSpecTemplate FlowPodSpec is for internal use?

Reply:
No, the `PodSpecTemplate` has an array of `containers` mapped by the container name. We set the application container to `workflow`. So if a user overrides the defaults, they should to change `spec.podSpecTemplate.containers[workflow].` That's cumbersome so we added a shortcut to `spec.flowContainer`. I'll clarify this in the OpenAPI + docs.

Reply:
ok, thanks for the explanation.